optaplanner-examples/src/main/java/org/optaplanner/examples/machinereassignment/solver/score/MachineReassignmentConstraintProvider.java

```diff
@@ -139,42 +145,40 @@ private Constraint transientUsage(ConstraintFactory factory) {
      * Load: The safety capacity for each resource for each machine should not be exceeded.
      */
     private Constraint loadCost(ConstraintFactory factory) {
-        throw new UnsupportedOperationException("Not yet implemented due to missing support for bi-grouping.");
-//        return factory.from(MrMachineCapacity.class)
-//                .join(MrProcessAssignment.class,
-//                        equal(MrMachineCapacity::getMachine, MrProcessAssignment::getMachine)
-//                )
-//                .groupBy(
-//                        (machineCapacity, processAssignment) -> machineCapacity, sumLong(
-//                                (machineCapacity, processAssignment) -> processAssignment.getUsage(machineCapacity.getResource())
-//                        )
-//                )
-//                .filter(((machineCapacity, usage) -> machineCapacity.getSafetyCapacity() < usage))
-//                .penalizeLong(MrConstraints.LOAD_COST,
-//                        HardSoftLongScore.ofSoft(1L),
-//                        (machineCapacity, usage) -> machineCapacity.getSafetyCapacity() - usage);
+        return factory.from(MrMachineCapacity.class)
+                .join(MrProcessAssignment.class,
+                      equal(MrMachineCapacity::getMachine, MrProcessAssignment::getMachine)
+                )
+                .groupBy((machineCapacity, processAssignment) -> machineCapacity, sumLong(
+                        (machineCapacity, processAssignment) -> processAssignment.getUsage(machineCapacity.getResource()))
```

Comment:
Any chance the `sumLong()` could come on a new line, like you do in other constraints?